source_middleware.go

```diff
@@ -34,3 +59,350 @@ func SourceWithMiddleware(d Source, middleware ...SourceMiddleware) Source {
 	}
 	return d
 }
+
+// -- SourceWithSchema ----------------------------------------------
+
+const (
+	configSourceSchemaType           = "sdk.schema.type"
+	configSourceSchemaPayloadEncode  = "sdk.schema.payload.encode"
+	configSourceSchemaPayloadSubject = "sdk.schema.payload.subject"
+	configSourceSchemaKeyEncode      = "sdk.schema.key.encode"
+	configSourceSchemaKeySubject     = "sdk.schema.key.subject"
+)
+
+// WithSourceWithSchemaConfig returns a SourceMiddlewareOption that sets the
+// default configuration for the SourceWithSchema middleware.
+func WithSourceWithSchemaConfig(cfg SourceWithSchemaConfig) SourceMiddlewareOption {
+	return func(m SourceMiddleware) {
+		if s, ok := m.(*SourceWithSchema); ok {
+			s.Config = cfg
```

Comment:
Having all the config in one struct makes it convenient because we need to provide only one `SourceMiddlewareOption`. Thinking out loud: do we want to make it possible to change only individual config fields (without having to copy paste a default config, and then modifying only 1 or 2 fields)? It would mean many more options, which isn't that nice.

Another option is maybe to have a default config exposed? Then a developer could simply take that, override some values and done.

Reply:
That's another reason why I went with having the empty value reflect the default behavior, because then we can get away with a single option to pass in the config and you can only set the fields you care about, all the others will still retain the default behavior.

Reply:
Gotcha, I re-read the `Wrap()` method just now and realized that. It would be nice to call it out in a comment or something, it's making it nice to customize a middleware!

Reply:
Ok, added a few comments in the config and `Wrap` to make this clearer 👍